app/src/common/shared/com/igalia/wolvic/ui/widgets/dialogs/LegalDocumentDialogWidget.java

```diff
@@ -50,24 +58,31 @@ public void updateUI() {
         LayoutInflater inflater = LayoutInflater.from(getContext());
 
         // Inflate this data binding layout
-        mBinding = DataBindingUtil.inflate(inflater, R.layout.privacy_policy_dialog, this, true);
+        mBinding = DataBindingUtil.inflate(inflater, R.layout.legal_document_dialog, this, true);
+
+        // Add the content of the specific legal document
+        if (mLegalDocument == LegalDocument.TERMS_OF_SERVICE) {
+            inflater.inflate(R.layout.terms_service_content, mBinding.scrollbar, true);
+        } else {
+            inflater.inflate(R.layout.privacy_policy_content, mBinding.scrollbar, true);
+        }
```

Comment:
What about using a switch? That way the compiler could warn us in case we were not handling any other legal document we might add in the future.

Reply:
Done.